src/JsonRpc/JsonRpc.csproj

```diff
@@ -7,6 +7,7 @@
         <PackageDescription>Primitives for working with JsonRpc.  This library is used as the base for communication with language servers</PackageDescription>
     </PropertyGroup>
     <ItemGroup>
+        <PackageReference Include="System.Collections.Immutable" />
```

Comment:
Curious what this is good for over `IReadOnly*` types.

Reply:
I grabbed it for ImmutableArray mainly because it "does the right thing" when it comes to creating new and transformed arrays.

I used this so I don't have to think too hard about copying arrays around.  This is because `SemanticTokensDocument` keeps a copy of the previous tokens, and the newly computed tokens, so that it can handle computing the diff between the two.

I was porting the builder from https://github.com/microsoft/vscode-languageserver-node/blob/master/server/src/semanticTokens.proposed.ts#L45 😄